python/ray/tests/test_node_labels.py

```diff
@@ -155,18 +143,21 @@ def test_autoscaler_set_node_labels(autoscaler_v2, shutdown_only):
 
 
 def test_ray_start_set_node_labels_from_file():
-    with tempfile.NamedTemporaryFile(mode="w+", delete=True) as test_file:
+    with tempfile.NamedTemporaryFile(mode="w+", delete=False) as test_file:
         test_file.write('"gpu_type": "A100"\n"region": "us"\n"market-type": "spot"')
-        test_file.flush()  # Ensure data is written
+        test_file_path = test_file.name
 
-        cmd = ["ray", "start", "--head", "--labels-file", test_file.name]
+    try:
+        cmd = ["ray", "start", "--head", "--labels-file", test_file_path]
         subprocess.check_call(cmd)
         ray.init(address="auto")
         node_info = ray.nodes()[0]
         assert node_info["Labels"] == add_default_labels(
             node_info, {"gpu_type": "A100", "region": "us", "market-type": "spot"}
         )
+    finally:
         subprocess.check_call(["ray", "stop", "--force"])
+        os.remove(test_file_path)
```

Comment:
will merge the PR as-is, but please follow up to move all such cleanup logic in this file into fixtures as best practice